src/database/models.py

```diff
@@ -4379,5 +4381,185 @@ def simple_json(self):
+        return res
+
+    def full_json(self):
+        return self.simple_json()
+
+    class Meta:
+        db_table = "custom_page_versions"
+        ordering = ("created_at",)
+
+
+
+class CustomPageBlock(BaseModel):
+    """
+        A class used to represent a custom page block(element) of a custom page on the MassEnergize platform
+        Blocks are the building blocks of a custom page. They can be of different types like text, image, video, section etc.
+    """
+    custom_page = models.ForeignKey(CustomPage, on_delete=models.CASCADE, db_index=True, related_name="page_sections")
+    order = models.IntegerField(default=0, blank=True)
+    props = models.JSONField(blank=True, null=True, default=dict)
+    direction = models.CharField(max_length=TINY_STR_LEN, blank=True)
+    type = models.CharField(max_length=SHORT_STR_LEN, blank=True)
+    text = models.TextField(blank=True, null=True)
+    parent = models.ForeignKey("self", on_delete=models.CASCADE, blank=True, null=True, related_name="children")
+    
+    def __str__(self):
+        return f"{self.custom_page.title} - {self.type}"
+    
+    def simple_json(self):
+        res = super().to_json()
+        res.update(model_to_dict(self))
+        res["content"] = [e.simple_json() for e in self.children.all()]
+        return res
+
+    def full_json(self):
+        return self.simple_json()
+
+    class Meta:
+        db_table = "custom_page_blocks"
+        ordering = ("order",)
         
+
+class CommunityCustomPage(BaseModel):
+    """
+    CommunityCustomPage model represents a custom page associated with a community as the owner of the page. 
+    The page can be shared with other communities and can be made public or private.
+
+    """
+    community = models.ForeignKey(Community, on_delete=models.CASCADE, db_index=True, related_name="custom_pages")
+    custom_page = models.ForeignKey(CustomPage, on_delete=models.CASCADE, db_index=True)
+    sharing_type =models.CharField(max_length=SHORT_STR_LEN, choices=SharingType.choices(), default=SharingType.OPEN.value[0])
+    audience = models.ManyToManyField(Community, blank=True, related_name="custom_pages_audience")
+
+
+    def __str__(self):
+        return f"{self.community.name} - {self.custom_page.title}"
+    
+    def simple_json(self):
+        res = super().to_json()
+        res.update(model_to_dict(self))
+        res["community"] = get_summary_info(self.community)
+        res["custom_page"] = get_summary_info(self.custom_page)
+        res["audience"] = [c.info() for c in self.audience.all()]
+
+        return res
+    
+    def full_json(self):
+        return self.simple_json()
+    
+    class Meta:
+        db_table = "community_custom_pages"
+        unique_together = ["community", "custom_page"]
+
+
+class CommunityCustomPageShare(BaseModel):
+    """
+    CommunityCustomPageShare model represents the relationship between a community and a custom page shared with that community.
+
+    """
```

Comment:
@abdullai-t Not sure the essence of this table. The `CommunityCustomPage` has a `Community` field so why create another table for this ? Are we trying to allow a `CommunityCustomPage` created by one community (e.g. Community A) with another community (e.g. Community B)